lib/containers/basetest.pm

```diff
@@ -0,0 +1,31 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Base class for container tests
+# Maintainer: qac team <qa-c@suse.de>
+
+package containers::basetest;
+use Mojo::Base 'opensusebasetest';
+use version_utils 'get_os_release';
+
+sub containers_factory {
+    my ($self, $runtime) = @_;
+    my $engine;
```

Comment:
It's only used in this single method, so let's keep it here.

Reply:
i will resolve it but maybe @cfconrad should have another look. because as the class is the base `$engine` should be defined for the whole instance. Then can be called in other functions directly. 
Anyway it is fine for now i guess

Reply:
I would keep it as @jlausuch did, as we do not use it in other methods inside of `container::basetest`.
If we will get a common `cleanup()` or other stuff, then I would go also with a member variable.